MIGRATION.md

```diff
@@ -53,13 +60,59 @@ Tip: Provide the `--transform`/`-t` argument at the end of the command, so that
 
 ## From v2.x to v3
 
-(intro)
+Circuit v3 is a large major release, including long-awaited changes from the full year that passed since v2. This guide will help you upgrade your application. Don't hesitate to contact the [maintainers](https://github.com/sumup-oss/circuit-ui#maintainers) if you have any further questions.
+
+### Accessibility
+
+Any accessible label props that were previously optional are now required and enforced in all components.
+
+To help identify places where accessible labels are missing (even if an app is built with TypeScript, some configurations will not break the build if required props are missing in JavaScript files), components will throw runtime errors in development at missing labels. Production and testing builds are not affected.
```

Comment:
The interjection about TypeScript disrupts the flow of the sentence and makes it more difficult to understand. I'd remove the interjection entirely or reword it to separate the what (throwing errors) and why (no type checks in JavaScript and imperfect type checks in TypeScript).